test/test_integration.py

```diff
@@ -237,14 +241,16 @@ def test_loading_slider(self):
             [Input('test-div', 'children')]
         )
         def delayed_value(children):
-            time.sleep(5)
+            lock.acquire()
+            lock.release()
```

Comment:
Very nice @Marc-Andre-Rivet !

@plotly/dash whenever you are tempted to add a `time.sleep` in tests, take a moment to think "is there a way to do it without a sleep?" A huge fraction of our test time is simply sleep statements, which are set super long in order to be robust.

A lot of the time we can solve this just by waiting for some condition (an element with the right contents, for example). This one is a bit trickier, we were using a sleep in the server process (where callbacks run) to ensure something had time to be observed in the browser-control process, but this commit shows a fairly simple pattern to do this with a `Lock` so that the server process stops waiting as soon as the browser has seen what it's looking for.